api/src/blocks.rs

```diff
@@ -89,11 +90,44 @@ impl BlocksApi {
         with_transactions: bool,
     ) -> BasicResultWith404<Block> {
         let latest_ledger_info = self.context.get_latest_ledger_info()?;
-        self.context.get_block_by_version(
-            &accept_type,
-            version,
-            latest_ledger_info,
-            with_transactions,
-        )
+        let bcs_block =
+            self.context
+                .get_block_by_version(version, &latest_ledger_info, with_transactions)?;
+
+        self.render_bcs_block(&accept_type, latest_ledger_info, bcs_block)
+    }
+
+    fn render_bcs_block(
+        &self,
+        accept_type: &AcceptType,
+        latest_ledger_info: LedgerInfo,
+        bcs_block: BcsBlock,
+    ) -> BasicResultWith404<Block> {
+        match accept_type {
+            AcceptType::Json => {
+                let transactions = if let Some(inner) = bcs_block.transactions {
+                    Some(
+                        self.context
+                            .render_transactions_sequential(inner, bcs_block.block_timestamp)?,
+                    )
+                } else {
+                    None
+                };
+                let block = Block {
+                    block_height: bcs_block.block_height.into(),
+                    block_hash: bcs_block.block_hash.into(),
+                    block_timestamp: bcs_block.block_timestamp.into(),
+                    first_version: bcs_block.first_version.into(),
+                    last_version: bcs_block.last_version.into(),
```

Comment:
do we actually need this field? last_version = first_version + len of transactions, no?

Reply:
You can request a block without the transactions for indexing purposes, then you need the last version.

Reply:
How do you get the len of transactions @davidiw? I had to infer len transactions from last and first version b/c that's all you get from the db call.